soroban-sdk/src/storage.rs

```diff
@@ -383,3 +458,49 @@ impl Instance {
         .unwrap_infallible();
     }
 }
+
+#[cfg(any(test, feature = "testutils"))]
+#[cfg_attr(feature = "docs", doc(cfg(feature = "testutils")))]
+impl testutils::storage::Instance for Instance {
+    fn all(&self) -> Map<Val, Val> {
+        let env = &self.storage.env;
+        let storage = env.host().with_mut_storage(|s| Ok(s.map.clone())).unwrap();
+        let address: xdr::ScAddress = env.current_contract_address().try_into().unwrap();
+        for entry in storage {
+            let (k, Some((v, _))) = entry else {
+                continue;
+            };
+            let xdr::LedgerKey::ContractData(xdr::LedgerKeyContractData { ref contract, .. }) = *k
+            else {
+                continue;
+            };
+            if contract != &address {
+                continue;
+            }
+            let xdr::LedgerEntry {
+                data:
+                    xdr::LedgerEntryData::ContractData(xdr::ContractDataEntry {
+                        key: xdr::ScVal::LedgerKeyContractInstance,
+                        val:
+                            xdr::ScVal::ContractInstance(xdr::ScContractInstance {
+                                ref storage, ..
+                            }),
+                        ..
+                    }),
+                ..
+            } = *v
+            else {
+                continue;
+            };
+            return match storage {
+                Some(map) => {
+                    let map: Val =
+                        Val::try_from_val(env, &xdr::ScVal::Map(Some(map.clone()))).unwrap();
```

Comment:
I really didn't want to introduce an ScMap clone in here, but it is too inconvenient to work around. I need it in an owned form for the ScVal unfortunately.